src/translations/en.json

```diff
@@ -1965,7 +1965,7 @@
                   "label": "State",
                   "attribute": "Attribute (optional)",
                   "from": "From (optional)",
-                  "for": "For (optional)",
+                  "for": "For",
```

Comment:
The only other place this translation was used is `ha-automation-condition-state.ts`, which is another form that only has `*` for required fields. This was the only field that showed `(optional)`, so this ended up being a good change on that form as well.